### PR TITLE
Add `assertions` to `ExportNamedDeclaration` without `from`

### PR DESCRIPTION
**Pull request by sosukesuzuki**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | N/A
| Patch: Bug Fix?          | Y
| Tests Added + Pass?      | Yes
| License                  | MIT

Add `assertions` to `ExportNamedDeclaration` without `from` when `importAssertions` is enabled.

According to [estree spec](https://github.com/estree/estree/blob/master/stage3/import-assertions.md#exportnameddeclaration):

> assertions must be an empty array when source is null.

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/13957"></a>

### CODE REVIEW

File: packages/babel-parser/src/parser/statement.js [hunk 1]
```diff
@@ -1824,6 +1824,9 @@ export default class StatementParser extends ExpressionParser {
 
     if (isFromRequired || hasSpecifiers || hasDeclaration) {
       this.checkExport(node, true, false, !!node.source);
+      if (this.hasPlugin("importAssertions") && node.assertions == null) {
+        node.assertions = [];
+      }
```
Comment (nicolo-ribaudo) [inline packages/babel-parser/src/parser/statement.js:1829]: I don't remember if our flow version supports it, but if it does we can do
```suggestion
      if (this.hasPlugin("importAssertions")) {
        node.assertions ??= [];
      }
```

**Review by existentialism — APPROVED** [review]

**Comment by nicolo-ribaudo** [comment]
Ok, it looks like it doesn't support it :grimacing:

**Review by JLHwung — CHANGES_REQUESTED** [review]
We can assign assertions immediately after we have parsed them:

https://github.com/babel/babel/blob/966387d263896a22913c7e1a511f5cc96b345621/packages/babel-parser/src/parser/statement.js#L2019

File: packages/babel-parser/src/parser/statement.js [hunk 2]
```diff
@@ -2005,12 +2011,8 @@ export default class StatementParser extends ExpressionParser {
       if (assertions) {
         node.assertions = assertions;
       }
-    } else {
-      if (expect) {
-        this.unexpected();
-      } else {
-        node.source = null;
```
Comment (sosukesuzuki) [inline packages/babel-parser/src/parser/statement.js:2012]: Actually, `node.source = null` is called in `maybeParseExportNamedSpecifiers` and `maybeParseExportDeclaration`. So we can remove this from here.

**Review by JLHwung — APPROVED** [review]

**Review by nicolo-ribaudo — APPROVED** [review]

**Pull request merged**